### PR TITLE
Add options to POWERONSTATE for only when device is actually powered on.

### PR DESCRIPTION
**Pull request by crankyoldgit**
TL;DR: All my lights etc. shouldn't turn on just because the WiFi router
reboots or I upload a new version of this software.

ESP8266s can tell if they were powered on or are restarted for numerous reasons.
Add POWERONSTATEs which are only acted upon if we detect the ESP8266 was
powered on, not just reset. Only makes sense if SAVESTATE is on.
e.g. '4' to turn on the relay only at power up, not just every restart.
     '5' to toggle the relay only at power up, not just every restart.

This solves an issue where the relay state is being changed if network
connection is lost.

Scenario:
Sonoff is in series between a device and a wall switch.
Desired action is to emulate traditional device behaviour when wall switch is
turned on. i.e. Physically present human can always turn on device.
If the switch is kept in on state, the device is left to be controlled by the
sonoff. When the sonoff loses network connection, or is updated etc. the
default action of 'POWERONSTATE 1' would be triggered, turning on the relay.
If the relay was previously off, the desired state would change.
'POWERONSTATE 4' preserves that state, but allows a human change the state
via the physical switch.

Note: MQTT Retain doesn't fit this scenario if POWER state was retained.
If the POWERONSTATE was 1, and the previous retained relay state was POWER OFF,
at device reset/restart (not just power on) a new 'POWER ON' state would be
transmitted and retained. Not the desired result.

### CODE REVIEW

**Comment by arendst** [comment]
Kind of integrated in version 3.2.8 just released (made Power On mandatory)

**Comment by crankyoldgit** [comment]
LGTM! Thanks.
